Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -64,11 +64,11 @@ def universal_archs
         [arch_64_bit, arch_32_bit].extend ArchitectureListExtension
       end
 
-      # True when running under an Intel-based shell via Rosetta on an
+      # True when running under an Intel-based shell via Rosetta 2 on an
       # Apple Silicon Mac. This can be detected via seeing if there's a
       # conflict between what `uname` reports and the underlying `sysctl` flags,
-      # since the `sysctl` flags don't change behaviour under Rosetta.
-      def in_rosetta?
+      # since the `sysctl` flags don't change behaviour under Rosetta 2.
+      def in_rosetta2?
         intel? && physical_cpu_arm64?
```

Comment:
It seems [Apple's recommended way to detect Rosetta 2](https://developer.apple.com/documentation/apple_silicon/about_the_rosetta_translation_environment) is to check `sysctl.proc_translated`.